flang/test/Semantics/OpenMP/map-clause.f90

```diff
@@ -2,16 +2,20 @@
 ! Check OpenMP MAP clause validity. Section 5.8.3 OpenMP 5.2.
 
 subroutine sb(arr)
+  implicit none
   real(8) :: arr(*)
   real :: a
-
+  integer:: b, c, i
+  common /var/ b, c  
+  
   !ERROR: Assumed-size whole arrays may not appear on the MAP clause
   !$omp target map(arr)
   do i = 1, 100
      a = 3.14
   enddo
   !$omp end target
 
+  !ERROR: Assumed-size array 'arr' must have explicit final subscript upper bound value
```

Comment:
Annoying ping for you @mjklemm I think it's correct for us to emit an error in this case (OpenMP section syntax falls-back on fortran rules from my understanding and we also have the following OpenMP line: "The upper bound for the last dimension of an assumed-size dummy array must be specified", which I think covers this case but my Fortran and OpenMP knowledge is admittedly lacking), but just incase I thought I'd pester you for a double check!

Reply:
This is fine.

Reply:
F2023 C930: "The second subscript shall not be omitted from a _subscript-triplet_ in the last dimension of an assumed-size array."

Reply:
> F2023 C930: "The second subscript shall not be omitted from a _subscript-triplet_ in the last dimension of an assumed-size array."

Would it fall into any other ruling (my appologies I don't have access to a Fortran specification document in this case) as this seems to also be emitted elsewhere for regular Fortran e.g.:

```
subroutine s6(x)
  integer :: x(*)
!ERROR: Assumed-size array 'x' must have explicit final subscript upper bound value
  x(:) = [1, 2, 3]
end
```

And if not, does the OpenMP ruling apply in this case (e.g. "The upper bound for the last dimension of an assumed-size dummy array must be specified")? 

Or is this syntax perfectly legal and should be supported by the compiler (even if other compilers seem to error out on it, although, considering they also seem to not be too friendly to the common block syntax, that might not be the best way to judge things)?

Reply:
As far as I know, the OpenMP array section syntax does not extend any Fortran array section syntax. The OpenMP spec only restricts where (clause or directive) the array section can appear. So I think the restriction, 

>"The second subscript shall not be omitted from a subscript-triplet in the last dimension of an assumed-size array."

is repeating C930. I will double check on it.

Reply:
Thank you very much @kkwli ! Just not wanting to accidentally increase restrictions incorrectly if it's avoidable.